File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math


def solution(progresses, speeds):
    answer = []
    days_left = [math.ceil((100 - progress) / speed) for progress, speed in zip(progresses, speeds)]

    print(days_left)  # 남은 일수를 확인

    current_day = days_left[0]
    count = 0

    for day in days_left:
        if day > current_day:
            answer.append(count)
            current_day = day
            count = 1
        else:
            count += 1

    answer.append(count)  # 마지막 배포 묶음 추가

    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        List<Integer> answer = new ArrayList<>();
        List<Integer> dayLeft = new ArrayList<>();

        for(int i = 0; i < progresses.length; i++) {
            dayLeft.add((int) Math.ceil((100.0 - progresses[i]) / speeds[i]));
        }

        int count = 1;
        int checkNumber = dayLeft.get(0);

        for(int i = 1; i < dayLeft.size(); i++) {
            // 다음 작업이 현재 작업보다 늦게 완료될 때
            if(checkNumber >= dayLeft.get(i)) {
                count++;
            } else {
                answer.add(count); // 현재 count를 저장
                count = 1; // 새로운 count 초기화
                checkNumber = dayLeft.get(i);
            }
        }

        // 마지막 남아 있는 count 추가
        answer.add(count);

        return answer.stream()
                .mapToInt(i -> i).toArray();
    }
}